File: src/behavioral/strategy/Strategy.cs

using System;

// Strategy interface
public interface IStrategy
{
    void Execute();
}

// Concrete Strategy A
public class ConcreteStrategyA : IStrategy
{
    public void Execute()
    {
        Console.WriteLine("ConcreteStrategyA is being executed.");
    }
}

// Concrete Strategy B
public class ConcreteStrategyB : IStrategy
{
    public void Execute()
    {
        Console.WriteLine("ConcreteStrategyB is being executed.");
    }
}

// Context class
public class Context
{
    private IStrategy _strategy;

    public void SetStrategy(IStrategy strategy)
    {
        _strategy = strategy;
    }

    public void ExecuteStrategy()
    {
        _strategy.Execute();
    }
}

class Program
{
    static void Main()
    {
        Context context = new Context();

        context.SetStrategy(new ConcreteStrategyA());
        context.ExecuteStrategy();

        context.SetStrategy(new ConcreteStrategyB());
        context.ExecuteStrategy();
    }
}

File: behavioral/strategy/strategy.py

# Strategy interface
class Strategy:
    def execute(self):
        pass

# Concrete Strategy A
class ConcreteStrategyA(Strategy):
    def execute(self):
        print("ConcreteStrategyA is being executed.")

# Concrete Strategy B
class ConcreteStrategyB(Strategy):
    def execute(self):
        print("ConcreteStrategyB is being executed.")

# Context class
class Context:
    def __init__(self):
        self._strategy = None

    def set_strategy(self, strategy):
        self._strategy = strategy

    def execute_strategy(self):
        self._strategy.execute()

if __name__ == "__main__":
    context = Context()

    context.set_strategy(ConcreteStrategyA())
    context.execute_strategy()

    context.set_strategy(ConcreteStrategyB())
    context.execute_strategy()
